nni/common/hpo_utils/dedup.py

```diff
@@ -52,29 +57,29 @@ def __call__(self, formatted_parameters):
             self._init_grid_search()
 
         while True:
-            new = self._grid_search._suggest()
+            new = self._grid_search._suggest()  # type: ignore
             if new is None:
                 raise nni.NoMoreTrialError()
             if self._not_dup(new):
                 return new
 
-    def _init_grid_search(self):
+    def _init_grid_search(self) -> None:
         from nni.algorithms.hpo.gridsearch_tuner import GridSearchTuner
         self._grid_search = GridSearchTuner()
         self._grid_search.history = self._history
         self._grid_search.space = self._space
         self._grid_search._init_grid()
 
-    def _not_dup(self, formatted_parameters):
+    def _not_dup(self, formatted_parameters: FormattedParameters) -> bool:
         params = deformat_parameters(formatted_parameters, self._space)
-        params_str = nni.dump(params, sort_keys=True)
+        params_str = typing.cast(str, nni.dump(params, sort_keys=True))
```

Comment:
@ultmaster I think it would be better to divide "dump to string" and "dump to bytes".

Reply:
I'll do that in another PR.